src/mactelnetd.c

```diff
@@ -456,7 +468,7 @@ static void user_login(struct mt_connection *curconn, struct mt_mactelnet_hdr *p
 		}
 	}
 
-	if (user == NULL || memcmp(md5sum, curconn->trypassword, 17) != 0) {
+	if (user == NULL || memcmp(hashsum, curconn->trypassword, 17) != 0) {
```

Comment:
At L157 `trypassword[17]` was changed into `trypassword[32]`, I guess this should be reflected here too.
Also need to make sure to zero-fill the rest of the buffer in L646.